pystac/collection.py

```diff
@@ -675,13 +675,32 @@ def from_dict(
 
         return collection
 
-    def get_assets(self) -> Dict[str, Asset]:
-        """Get this item's assets.
+    def get_assets(
+        self,
+        media_type: Optional[Union[str, pystac.MediaType]] = None,
+        role: Optional[Union[str, pystac.MediaType]] = None,
+    ) -> Dict[str, Asset]:
+        """Get this collection's assets.
+
+        Args:
+            media_type: If set, filter the assets such that only those with a
+                matching ``media_type`` are returned.
+            role: If set, filter the assets such that only those with a matching
+                ``role`` are returned.
 
         Returns:
-            Dict[str, Asset]: A copy of the dictionary of this item's assets.
+            Dict[str, Asset]: A dictionary of assets that match ``media_type``
+                and/or ``role`` if set or else all of this collection's assets.
         """
-        return dict(self.assets.items())
+        if media_type is None and role is None:
+            return dict(self.assets.items())
+        assets = dict()
+        for key, asset in self.assets.items():
+            if (media_type is None or asset.media_type == media_type) and (
+                role is None or (asset.roles is not None and role in asset.roles)
```

Comment:
I agree that this is getting a little wild. What do you think about a new `Asset.has_role` method? E.g.

```suggestion
            if (media_type is None or asset.media_type == media_type) and (role is None or asset.has_role(role)):
```

and then at https://github.com/stac-utils/pystac/blob/017ddfd8592e822ecc6905b5ea3ed0cd121798dc/pystac/asset.py:

```python
def has_role(self, role: str) -> bool:
    if self.roles is None:
        return False
    else:
        return role in self.roles

Reply:
Yeah, that works for me. Done.